### PR TITLE
 MAILBOX-327 Adding and firing QuotaEvents 

### PR DESCRIPTION
**Pull request by chibenwa**
Only last commit.

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java [hunk 1]
```diff
@@ -57,6 +62,56 @@
      */
     void event(Event event);
 
+    class QuotaEvent implements Event, Serializable {
```
Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: I don't think every quota events are about the value of quotas for a given quotaroot.
I would make another level in the hierarchy and name this one QuotaRootUsageUpdateEvent

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 2]
```diff
@@ -915,4 +920,33 @@ public void getMetaDataShouldReturnDefaultValueWhenNoReadRight() throws MailboxE
...
+        mailboxManager.addGlobalListener(listener, session);
+
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+        mailboxManager.getMailbox(inbox, session)
+            .appendMessage(MessageManager.AppendCommand.builder()
+                .build("header: value\r\nbody"), session);
+
+        assertThat(listener.getEvents())
+            .filteredOn(event -> event instanceof MailboxListener.QuotaEvent)
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: why do you want to filter when you could call `contains` instead ?

**Comment by chibenwa** [comment]
test this please

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java [hunk 3]
```diff
@@ -56,6 +61,61 @@
      *            not null
      */
     void event(Event event);
+    
+    interface QuotaEvent extends Event {
+        QuotaRoot getQuotaRoot();
+    }
+
+    class QuotaUsageUpdateEvent implements QuotaEvent, Serializable {
```
Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: `Updated` IMO, event are from the past. Not sure it's consistent with existing code.

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 4]
```diff
@@ -915,4 +920,32 @@ public void getMetaDataShouldReturnDefaultValueWhenNoReadRight() throws MailboxE
...
+        session = mailboxManager.createSystemSession(USER_1);
+
+        EventCollector listener = new EventCollector();
+        mailboxManager.addGlobalListener(listener, session);
+
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+        mailboxManager.getMailbox(inbox, session)
+            .appendMessage(MessageManager.AppendCommand.builder()
+                .build("header: value\r\nbody"), session);
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: should not use String builder, right ?

Comment (chibenwa) [reply mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: Need to be rebased, yes.

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java (hunk 3 above)
Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java]: That was your proposal (cf comment above) but yes, ok.

**Comment by chibenwa** [comment]
merged

**Pull request closed without merging**